frontend/src/CreateDiagram/CreateDiagram.jsx

```diff
@@ -2,27 +2,51 @@ import React, { useState } from "react";
 
 import { Link } from "react-router-dom";
 
+import "./CreateDiagram.scss";
+
 export default function CreateDiagram({ onSubmit }) {
   const [diagramName, setDiagramName] = useState("");
 
   function submit() {
     onSubmit({ diagramName });
   }
 
+  function isValid() {
+    // TODO: use a regex for this
+    if (diagramName.includes("-")) {
+      return false;
+    }
+    if (diagramName.length < 1) {
+      return false;
+    }
+    return true;
+  }
+
+  function onDiagramNameChange(e) {
+    const newText = e.target.value.replace(/[^a-z0-9_]/gi, "");
+    setDiagramName(newText);
+  }
+
   return (
-    <div>
+    <div
+      className="create-diagram
+    "
+    >
       <Link to="/">
         <button className="home">Home</button>
       </Link>
-      <br />
-      <label>Create diagram</label>
-      <br />
-      <input
-        value={diagramName}
-        onChange={(e) => setDiagramName(e.target.value)}
-      />
-      <br />
-      <button onClick={submit}>Create</button>
+
+      <p className="label">
+        <span>Choose a diagram name</span>
+        <span className="explanation">
+          (only numbers, letters and underscores allowed)
```

Comment:
in a previous discussion, you said you wanted to allow "/", but I think, as we discussed earlier, that you change this in the next PR.